includes/class-wc-calypso-bridge-coming-soon.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+use Automattic\WooCommerce\Admin\Features\Features;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Calypso_Bridge_Coming_Soon
+ *
+ * @since   x.x.x
+ * @version x.x.x
+ *
+ * Handle Coming Soon mode.
+ */
+class WC_Calypso_Bridge_Coming_Soon {
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var object
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Get class instance.
+	 *
+	 * @return object Instance.
+	 */
+	final public static function get_instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'a8c_show_coming_soon_page', array( $this, 'should_show_a8c_coming_soon_page' ), 99999, 1 );
+	}
+
+	/**
+	 * Show the coming soon page if the Launch Your Store feature is enabled.
+	 *
+	 * @param bool $should_show
+	 * @return bool
+	 */
+	public function should_show_a8c_coming_soon_page( $should_show ) {
+		if ( ! Features::is_enabled( 'launch-your-store' ) ) {
```

Comment:
I forgot, there was some instances where `Features` class was not available during a run. I'm unsure how early `should_show_a8c_coming_soon_page` is run, but the prior art is to check for the class too: https://github.com/Automattic/wc-calypso-bridge/blob/3233649621c32b4f7f18daf68ee381f90a7df970/includes/class-wc-calypso-bridge-ecommerce-admin-menu.php#L380

What do you think?

Reply:
That's a good idea. Updated in  3038da3be2d72c9b5866676a94289233cbb528bc

Reply:
I wonder if there is anything we can do to prevent any future issues like this.  🤔  We also use `Features::is_enabled` directly here, though it should be safe to use it as it is in the setup task context. In other places, we use methods/classes from WooCommerce core without checking their availability. 

https://github.com/Automattic/wc-calypso-bridge/blob/717d11285b7b770ee7dbbc6e8251d7646ade4dfc/includes/class-wc-calypso-bridge-setup-tasks.php#L147-L153

  I'd also like to know why the Features class was unavailable during the run.

Reply:
Good question, @chihsuan! I think it's one of the pitfalls of WordPress' filters. I believe in some cases the filter ran before WooCommerce initialized.

I think a good solution might be to create a class helper that runs a class callback after WooCommerce has initialized, but I'm not sure how that'll look

Reply:
To be accurate, it's very much possible that the `should_show_a8c_coming_soon_page` filter is safe. I was just trying to be guarantee a safe execution, since it's possible for other mu-plugins to do a `apply_filter( 'should_show_a8c_coming_soon_page' )` before WooCommerce is loaded

Reply:
Good point! Agreed that it's better to be safe.